### PR TITLE
feat(consumer-restriction): add route_id type support

### PR DESCRIPTION
**Pull request by jagerzhang**
### What this PR does / why we need it:
Fix https://github.com/apache/apisix/issues/4755

### CODE REVIEW

File: apisix/plugins/consumer-restriction.lua [hunk 1]
```diff
@@ -70,6 +70,9 @@ local _M = {
 }
 
 local fetch_val_funcs = {
+    ["route_id"] = function(ctx)
```
Comment (spacewander) [inline apisix/plugins/consumer-restriction.lua:73]: Test required.

Comment (jagerzhang) [reply apisix/plugins/consumer-restriction.lua:73]: 很奇怪，我搭建的测试环境跑官方的用例会报错，看错误提示是因为前面有用例没执行，导致后面的用例返回404.. 很奇怪为啥会跳过前面加路由的用例。

单独拷贝测试用例都能pass，说明我的测试环境应该没问题，但是多个用例组合一起，总是发现前面的用例不运行，直接跳过了。。

image

Comment (spacewander) [reply apisix/plugins/consumer-restriction.lua:73]: Could you paste test3.t here? Should I reopen this PR so you can continue to work on it?

**Comment by jagerzhang** [comment]
already add test for this feature, Plz review it.

File: apisix/plugins/consumer-restriction.lua (hunk 1 above)
Comment (jagerzhang) [reply apisix/plugins/consumer-restriction.lua:73]: already add test for this feature, Plz review it.

**Review by tokers — APPROVED** [review]

**Review by starsz — APPROVED** [review]

File: t/plugin/consumer-restriction.t [hunk 2]
```diff
@@ -1476,7 +1476,242 @@ passed
...
+--- no_error_log
+[error]
+
+
+
+=== TEST 48: verify: valid whitelist `route_id`
+--- config
+location /t {
+    content_by_lua_block {
+        local ngx_time = ngx.time
```
Comment (tzssangglass) [inline t/plugin/consumer-restriction.t:1540]:
```suggestion
        local ngx_time = ngx.time()
```

File: t/plugin/consumer-restriction.t [hunk 3]
```diff
@@ -1476,7 +1476,242 @@ passed
...
+        local ngx_http_time = ngx.http_time
+        local core = require("apisix.core")
+        local t = require("lib.test_admin")
+        local hmac = require("resty.hmac")
+        local ngx_encode_base64 = ngx.encode_base64
+
+        local secret_key = "my-secret-key"
+        local timestamp = ngx_time()
+        local gmt = ngx_http_time(timestamp)
```
Comment (tzssangglass) [inline t/plugin/consumer-restriction.t:1549]:
```suggestion
        local gmt = ngx_http_time(ngx_time)
```

**Review by spacewander — APPROVED** [review]

File: t/plugin/consumer-restriction.t (hunk 2 above)
Comment (jagerzhang) [reply t/plugin/consumer-restriction.t]: Modified

File: t/plugin/consumer-restriction.t (hunk 3 above)
Comment (jagerzhang) [reply t/plugin/consumer-restriction.t]: Modified

**Review by tzssangglass — APPROVED** [review]

**Pull request merged**